pkg/metallb/frr.go

```diff
@@ -0,0 +1,549 @@
+		builder.Definition.Name, builder.Definition.Namespace,
+	)
+
+	if !builder.Exists() {
+		return builder, fmt.Errorf("FrrConfiguration cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.Resource(
+		GetFrrConfigurationGVR()).Namespace(builder.Definition.Namespace).Delete(
+		context.TODO(), builder.Definition.Name, metav1.DeleteOptions{})
+
+	if err != nil {
+		return builder, fmt.Errorf("can not delete FRRConfiguration: %w", err)
+	}
+
+	builder.Object = nil
+
+	return builder, nil
+}
+
+// GetFrrConfigurationGVR returns FrrConfiguration GroupVersionResource which could be used for Clean function.
+func GetFrrConfigurationGVR() schema.GroupVersionResource {
+	return schema.GroupVersionResource{
+		Group: FRRAPIGroup, Version: APIVersion, Resource: "frrconfigurations",
+	}
+}
+
+// Exists checks whether the given FrrConfiguration exists.
+func (builder *FrrConfigurationBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof(
+		"Checking if FrrConfiguration %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// WithBGPRouter defines a BGP router as localASN.
+func (builder *FrrConfigurationBuilder) WithBGPRouter(localASN uint32) *FrrConfigurationBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Define BGP router with an ASN: %v", localASN)
+
+	if strconv.Itoa(int(localASN)) == "" {
+		glog.V(100).Infof("The local ASN of the frrConfiguration can not be empty")
+
+		builder.errorMsg = "frrConfiguration 'local AS' cannot be empty"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	// Ensure there is at least one router in the list
+	if len(builder.Definition.Spec.BGP.Routers) == 0 {
+		// If no routers exist, create one with the provided localASN
+		builder.Definition.Spec.BGP.Routers = append(builder.Definition.Spec.BGP.Routers, frrtypes.Router{
+			ASN: localASN,
+		})
+	} else {
+		// If a router exists at index 0, set its ASN to the provided localASN
+		builder.Definition.Spec.BGP.Routers[0].ASN = localASN
+	}
+
+	return builder
+}
+
+// WithToReceiveModeAll allows all incoming routes.
+func (builder *FrrConfigurationBuilder) WithToReceiveModeAll(neighborIndex int) *FrrConfigurationBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Allows the FRR nodes to receive all routes in namespace %s with mode type: all",
+		builder.Definition.Namespace)
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	// Check if BGP routers exist
+	if len(builder.Definition.Spec.BGP.Routers) == 0 {
+		builder.errorMsg = "No BGP routers defined."
+
+		return builder
+	}
+
+	// Access the first router (assuming it's hardcoded to index 0)
+	router := &builder.Definition.Spec.BGP.Routers[0]
```

Comment:
remove and use `builder.Definition.Spec.BGP.Routers[0].Neighbors` as you're doing it here
https://github.com/openshift-kni/eco-goinfra/pull/618/files#diff-b003dccc27e9d02986b5624edaadd459e75846819d4c7079f3d0d0529ec1cf80R267

Reply:
changed